libraries/standard/http/src/private/http_client_internal.h

```diff
@@ -85,4 +85,24 @@
 #define HTTP_CONNECTION_CLOSE_VALUE             "close"
 #define HTTP_CONNECTION_CLOSE_VALUE_LEN         ( sizeof( HTTP_CONNECTION_CLOSE_VALUE ) - 1 )
 
+
+/**
+ * @brief Constants relating to Range Requests.
+ */
+#define RANGE_REQUEST_HEADER_FIELD               "Range"
+#define RANGE_REQUEST_HEADER_FIELD_LEN           ( sizeof( RANGE_REQUEST_HEADER_FIELD ) - 1 )
+#define RANGE_REQUEST_HEADER_VALUE_PREFIX        "bytes="
+#define RANGE_REQUEST_HEADER_VALUE_PREFIX_LEN    ( sizeof( RANGE_REQUEST_HEADER_VALUE_PREFIX ) - 1 )
+
+/* Maximum value of a 32 bit signed integer is 2,147,483,647. Used for calculating buffer space for
+ * ASCII representation of range values. */
+#define MAX_INT32_NO_OF_DIGITS                   10
```

Comment:
This should be named `MAX_INT32_NO_OF_DECIMAL_DIGITS` since you need to know the base to know how long the string representation could be.  Also, what if the number is negative and has a leading `-`?  Not that this would happen in normal use, however one could see it happening in the presence of some other bug.

Reply:
Good suggestion about macro renaming. 
As you mentioned, the correct use case of negative value should would only contain a single value in the range as `"Range: bytes=<negative-value>"`. For a misuse of introducing negative value in the `[rangeStart, rangeEnd]` use case, I think the unit tests for `AddRangeHeader` would catch it...